code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -110,12 +110,20 @@
 		if(I_HARM)
 			//Vampire code
 			if(M.mind && M.mind.vampire && (M.mind in ticker.mode.vampires) && !M.mind.vampire.draining && M.zone_sel && M.zone_sel.selecting == "head" && src != M)
+				if(species && species.flags & NO_BLOOD)//why this hell were we never checkinf for this?
+					to_chat(M, "<span class='warning'>They have no blood!</span>")
 				if((head && (head.flags & HEADCOVERSMOUTH)) || (wear_mask && (wear_mask.flags & MASKCOVERSMOUTH)))
 					to_chat(M, "<span class='warning'>Remove their mask!</span>")
 					return
 				if((M.head && (M.head.flags & HEADCOVERSMOUTH)) || (M.wear_mask && (M.wear_mask.flags & MASKCOVERSMOUTH)))
-					to_chat(M, "<span class='warning'>Remove your mask!</span>")
-					return
+					if(M.get_species() == "Vox")
+						M.mind.vampire.handle_bloodsucking(src)
+						add_logs(src, M, "vampirebit")
+						msg_admin_attack("[key_name_admin(M)] vampirebit [key_name_admin(src)]")
+						return

```

Comment:
You don't need to copy the entire bloodsucking section - just have it skip the "remove your mask" return, and the other checks will hold up fine